packages/code-utils/src/index.ts

```diff
@@ -1,53 +1,72 @@
 import parseOpcode from "./opcodes";
-import type { Instruction, OpcodeTable, opcodeObject, opcodes } from "./types";
-export type { Instruction, OpcodeTable, opcodeObject, opcodes };
+import type { Instruction, OpcodeTable, DisassemblyOptions } from "./types";
+export type { Instruction, OpcodeTable, DisassemblyOptions };
 import * as cbor from "cbor";
 
 /**
  * parseCode - return a list of instructions given a 0x-prefixed code string.
  *
- * If numInstructions is not passed in, we attempt to strip contract
- * metadata.  This won't work very well if the code is for a constructor or a
- * contract that can create other contracts, but it's better than nothing.
+ * The optional second options argument allows two options; both are ways of
+ * attempting to limit the disassembly to only the code section rather than the
+ * data section.  If maxInstructionCount is used, the disassembly will be limited
+ * to the specified number of instructions (one may pass in here the number of
+ * instructions in the corresponding source map).
  *
- * WARNING: Don't invoke the function that way if you're dealing with a
- * constructor with arguments attached!  Then you could get disaster!
+ * If attemptStripMetadata is used, we will attempt to strip the metadata at the
+ * end of the code.  This is not reliable, and should be avoided if better
+ * alternatives are available.  It may be particularly unreliable when dealing with
+ * constructors that have had arguments attached to the end!
  *
- * If you pass in numInstructions (hint: count the semicolons in the source
- * map, then add one) this is used to exclude metadata instead.
+ * These options can be combined, although I'm not sure why you'd want to.
  *
  * @param  {String} hexString Hex string representing the code
  * @return Array               Array of instructions
  */
 export function parseCode(
   hexString: string,
-  numInstructions: number = null
+  { maxInstructionCount, attemptStripMetadata }: DisassemblyOptions = {}
```

Comment:
Do we care about breaking changes affecting external users of this package? I think not, but I'm not certain.

On the off chance that we do, would it be impossible or a bad idea to provide an overloaded signature so it's not a breaking change?

Reply:
Yeah, I figured it was probably OK to just break this and that the extra work to make it nonbreaking wasn't worth it... but that's what review is for!  I should note that I discussed this with @gnidan earlier and he thought it was OK to just break it, so that's at least one more person in favor of just breaking it. :)

Reply:
I mean this very much looks to be an internal package (ideally we wouldn't publish those 😬), so I certainly have no qualms. Just wanted to make sure it was a conscious choice.

Reply:
Oh yeah as I mentioned above, it had been my intent for a while to make that breaking change; I just hadn't really had any reason to touch this code anytime recently.  I figured I'd end up doing it when Shanghai hit and I needed to rewrite large chunks of this, but since I ended up going over this code *now* I figured I'd do it now instead!